mod/groups/actions/groups/membership/join.php

```diff
@@ -9,74 +10,88 @@
  *
  * @package ElggGroups
  */
-
-global $CONFIG;
-
 $user_guid = get_input('user_guid', elgg_get_logged_in_user_guid());
 $group_guid = get_input('group_guid');
 
-$user = get_user($user_guid);
-
-// access bypass for getting invisible group
-$ia = elgg_set_ignore_access(true);
-$group = get_entity($group_guid);
-elgg_set_ignore_access($ia);
-
-if ($user && ($group instanceof ElggGroup)) {
-
-	// join or request
-	$join = false;
-	if ($group->isPublicMembership() || $group->canEdit($user->guid)) {
-		// anyone can join public groups and admins can join any group
-		$join = true;
-	} else {
-		if (check_entity_relationship($group->guid, 'invited', $user->guid)) {
-			// user has invite to closed group
-			$join = true;
-		}
+$user = get_entity($user_guid, true);
```

Comment:
FYI, we should stop assuming that user entities are always accessible. I have implemented a few crazy access hacks on several projects that make user entities invisible until certain conditions are met.